### PR TITLE
Use Makefile.common from Prometheus

### LINKED ISSUE #1338: Use the main common makefile
https://github.com/prometheus/prometheus/blob/master/Makefile.common

### PR DESCRIPTION
**Pull request by simonpasquier**
Closes #1338

### CODE REVIEW

**Comment by simonpasquier** [comment]
@mxinden correct although `.PHONY` needs to include some targets that are only defined in `Makefile`.

File: Makefile.common [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+promu:
+	GOOS= GOARCH= $(GO) get -u github.com/prometheus/promu
+
+$(FIRST_GOPATH)/bin/staticcheck:
+	GOOS= GOARCH= $(GO) get -u honnef.co/go/tools/cmd/staticcheck
+
+$(FIRST_GOPATH)/bin/govendor:
+	GOOS= GOARCH= $(GO) get -u github.com/kardianos/govendor
+
+.PHONY: all style check_license format build test vet assets tarball docker promu staticcheck $(FIRST_GOPATH)/bin/staticcheck govendor $(FIRST_GOPATH)/bin/govendor
```
Comment (mxinden) [inline Makefile.common]: `govendor` is mentioned here, but I don't see a make target for it. Am I missing something?

Comment (simonpasquier) [reply Makefile.common]: It's defined on L97.

Comment (simonpasquier) [reply Makefile.common]: Pinging @krasi-georgiev but AFAICT it has been carried from the original Prometheus [Makefile](https://github.com/prometheus/prometheus/blob/f63e7db4cbdb616337ca877b306b9b96f7f4e381/Makefile#L106). I see no reason  to keep it but the change has to happen in prometheus/prometheus first.

[the remark this answers is not in the document]
Comment (simonpasquier) [reply Makefile.common]: Yes this is my understanding.

**Pull request merged**